crates/rspack_error/src/emitter.rs

```diff
@@ -175,34 +175,32 @@ fn emit_diagnostic<T: Write + WriteColor>(
   pwd: impl AsRef<Path>,
   files: &mut SimpleFiles<String, String>,
 ) -> crate::Result<()> {
-  if let Some(info) = &diagnostic.source_info {
-    let file_path = Path::new(&info.path);
-    let relative_path = file_path.relative(&pwd);
-    let relative_path = relative_path.as_os_str().to_string_lossy().to_string();
-    let file_id = files.add(relative_path, info.source.clone());
-    let diagnostic = Diagnostic::new(diagnostic.severity.into())
-      .with_message(&diagnostic.title)
-      // Because we don't have error code now, and I don't think we have
-      // enough energy to matain error code either in the future, so I use
-      // this field to represent diagnostic kind, looks pretty neat.
-      .with_code(diagnostic.kind.to_string())
-      .with_labels(vec![Label::primary(
-        file_id,
-        diagnostic.start..diagnostic.end,
+  let (labels, message) = match &diagnostic.source_info {
+    Some(info) => {
+      let file_path = Path::new(&info.path);
+      let relative_path = file_path.relative(&pwd);
+      let relative_path = relative_path.as_os_str().to_string_lossy().to_string();
+      let file_id = files.add(relative_path, info.source.clone());
+      (
+        vec![Label::primary(file_id, diagnostic.start..diagnostic.end)
+          .with_message(&diagnostic.message)],
+        diagnostic.title.clone(),
       )
-      .with_message(&diagnostic.message)]);
-
-    let config = codespan_reporting::term::Config::default();
-
-    term::emit(writer, &config, files, &diagnostic).expect("TODO:");
-  } else {
-    let color = match diagnostic.severity {
-      crate::Severity::Error => Color::Red,
-      crate::Severity::Warn => Color::Yellow,
-    };
-    writer.set_color(ColorSpec::new().set_fg(Some(color)))?;
```

Comment:
@IWANABETHATGUY  I don't understand why use separate emitter logic here and think it would be better unify

Reply:
The color is lost according to the screenshot.

Reply:
![image](https://github.com/web-infra-dev/rspack/assets/8898718/357de0c8-0671-49a8-a1cb-2ad7a382065e)
cargo also only render diagnostic.code to red and keep message as white